crates/store/re_dataframe/src/range.rs

```diff
@@ -0,0 +1,265 @@
+                re_tracing::profile_scope!("gather pov timestamps");
+
+                let query = RangeQuery::new(self.query.timeline, self.query.time_range)
+                    .keep_extra_timelines(true) // we want all the timelines we can get!
+                    .keep_extra_components(false);
+
+                let results = self.engine.cache.range(
+                    self.engine.store,
+                    &query,
+                    &self.query.pov.entity_path,
+                    [self.query.pov.component_name],
+                );
+
+                results
+                    .components
+                    .into_iter()
+                    .find_map(|(component_name, chunks)| {
+                        (component_name == self.query.pov.component_name).then_some(chunks)
+                    })
+                    .map(Into::into)
+            };
+
+            RangeQuerytHandleState {
+                columns,
+                pov_chunks,
+            }
+        })
+    }
+
+    /// All results returned by this handle will strictly follow this schema.
+    ///
+    /// Columns that do not yield any data will still be present in the results, filled with null values.
+    pub fn schema(&self) -> &[ColumnDescriptor] {
+        &self.init().columns
+    }
+
+    /// Partially executes the range query until the next natural page of results.
+    ///
+    /// Returns a single [`RecordBatch`] containing as many rows as available in the page, or
+    /// `None` if all the dataset has been returned.
+    /// Each cell in the result corresponds to the latest known value at that particular point in time
+    /// for each respective `ColumnDescriptor`.
+    ///
+    /// The schema of the returned [`RecordBatch`] is guaranteed to match the one returned by
+    /// [`Self::schema`].
+    /// Columns that do not yield any data will still be present in the results, filled with null values.
+    ///
+    /// "Natural pages" refers to pages of data that match 1:1 to the underlying storage.
+    /// The size of each page cannot be known in advance, as it depends on unspecified
+    /// implementation details.
+    /// This is the most performant way to iterate over the dataset.
+    ///
+    /// ```ignore
+    /// while let Some(batch) = query_handle.next_page() {
+    ///     // …
+    /// }
+    pub fn next_page(&mut self) -> Option<RecordBatch> {
+        re_tracing::profile_function!(format!("{:?}", self.query));
+
+        _ = self.init();
+        let pov_chunk = self.state.get_mut()?.pov_chunks.as_mut()?.pop_front()?;
+        let pov_time_column = pov_chunk.timelines().get(&self.query.timeline)?;
+        let columns = self.schema();
+
+        // TODO(cmc): There are more efficient, albeit infinitely more complicated ways to do this.
+        // Let's first implement all features (multi-PoV, pagination, timestamp streaming, etc) and
+        // see if this ever becomes an issue before going down this road.
```

Comment:
Makes sense. We'll always need this as a worst-case fallback path, but I can definitely imagine this being replaced with a series of checks that determine whether particular happy-path join operations are applicable with more efficient implementations for those cases.